src/Sylius/Behat/Page/Admin/Product/Common/ProductMediaTrait.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Page\Admin\Product\Common;
+
+use Behat\Mink\Element\NodeElement;
+use Behat\Mink\Exception\ElementNotFoundException;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+trait ProductMediaTrait
+{
+    /** @var string[] */
+    private array $imageUrls = [];
+
+    public function getDefinedProductMediaElements(): array
+    {
+        return [
+            'add-image' => '[data-test-add-image]',
+            'image_subform_with_type' => '[data-test-image-subform][data-test-type="%type%"]',
+            'images' => '[data-test-images]',
+        ];
+    }
+
+    public function attachImage(string $path, ?string $type = null, ?ProductVariantInterface $productVariant = null): void
+    {
+        $this->changeTab('media');
+        $this->getElement('add-image')->click();
+
+        $this->waitForFormUpdate();
+
+        $images = $this->getElement('images');
+        $imagesSubform = $images->findAll('css', '[data-test-image-subform]');
+        $imageSubform = end($imagesSubform);
+
+        if (null !== $type) {
+            $imageSubform->fillField('Type', $type);
+        }
+
+        if (null !== $productVariant) {
+            $this->autocompleteHelper->selectByValue(
+                $this->getDriver(),
+                $imageSubform->find('css', '[data-test-product-variant]')->getXpath(),
+                $productVariant->getCode(),
+            );
+        }
+
+        $filesPath = $this->getParameter('files_path');
+        $imageSubform->find('css', '[data-test-file]')->attachFile($filesPath . $path);
+    }
+
+    public function hasLastImageAVariant(ProductVariantInterface $productVariant): bool
+    {
+        return $this->hasImageWithVariant($productVariant);
+    }
+
+    public function changeImageWithType(string $type, string $path): void
+    {
+        $filesPath = $this->getParameter('files_path');
+
+        $imageSubform = $this->getElement('image_subform_with_type', ['%type%' => $type]);
+        $imageSubform->find('css', '[data-test-file]')->attachFile($filesPath . $path);
+    }
+
+    public function removeImageWithType(string $type): void
+    {
+        $this->changeTab('media');
+
+        $imageSubform = $this->getElement('image_subform_with_type', ['%type%' => $type]);
+        $imageSubform->find('css', '[data-test-image-delete]')->click();
+        $this->waitForFormUpdate();
+    }
+
+    public function removeFirstImage(): void
+    {
+        $this->changeTab('media');
+        $firstSubform = $this->getFirstImageSubform();
+        $firstSubform->findAll('css', '[data-test-image-delete]')[0]->click();
```

Comment:
Since it's an image subform I'd assume there'll only be one delete button.
```suggestion
        $firstSubform = $this->getFirstImageSubform();
        $firstSubform->find('css', '[data-test-image-delete]')->click();
```

Reply:
https://github.com/Sylius/Sylius/pull/16349/files#diff-fd91c9ddf0b416af72abdd28aad3bd674c01da691affeb9de436c9df50abfb3dR84